cider/exec/template/NativeCodegen.cpp

```diff
@@ -1722,27 +1715,33 @@ Executor::compileWorkUnit(const std::vector<InputTableInfo>& query_infos,
       optimize_ir(query_func, cgen_state_->module_, pass_manager, live_funcs, co);
 #endif  // WITH_JIT_DEBUG
     }
-    llvm_ir =
-        serialize_llvm_object(multifrag_query_func) + serialize_llvm_object(query_func) +
-        serialize_llvm_object(cgen_state_->row_func_) +
-        (cgen_state_->filter_func_ ? serialize_llvm_object(cgen_state_->filter_func_)
-                                   : "");
+    if (eo.just_explain == EXPLAIN_FUNC) {
+      llvm_ir =
+          serialize_llvm_object(multifrag_query_func) +
+          serialize_llvm_object(query_func) +
+          serialize_llvm_object(cgen_state_->row_func_) +
+          (cgen_state_->filter_func_ ? serialize_llvm_object(cgen_state_->filter_func_)
+                                     : "");
 
 #ifndef NDEBUG
-    llvm_ir += serialize_llvm_metadata_footnotes(query_func, cgen_state_.get());
+      llvm_ir += serialize_llvm_metadata_footnotes(query_func, cgen_state_.get());
 #endif
+    } else if (eo.just_explain == EXPLAIN_MODULE) {
+      llvm_ir = serialize_llvm_object(cgen_state_->module_);
+    }
   }
 
   LOG(IR) << "\n\n" << query_mem_desc->toString() << "\n";
-  LOG(DEBUG1) << "IR for the CPU: \n";
 #ifndef NDEBUG
+  LOG(DEBUG1) << "IR for the CPU: \n";
```

Comment:
Just remove this since we're only CPU?

Reply:
I think it's also used for the header of IR in the log